wom/src/main/scala/wom/types/WomLongType.scala

```diff
@@ -0,0 +1,18 @@
+package wom.types
+
+import spray.json.{JsNumber, JsString}
+import wom.values.{WomInteger, WomLong, WomString}
+
+case object WomLongType extends WomPrimitiveType {
+  val toDisplayString: String = "Long"
+
+  override protected def coercion = {
+    case i: Long => WomLong(i)
+    case i: Integer => WomLong(i.toLong)
+    case n: JsNumber if n.value.isValidLong => WomInteger(n.value.intValue())
```

Comment:
That won't end well if it doesn't fit in an int. And shouldn't all this stuff end up as `WomLong`s?

Reply:
ToL:  some unit tests to validate those coercions would be nice too